lib/workers/repository/finalise/prune.ts

```diff
@@ -93,7 +93,7 @@ export async function pruneStaleBranches(
   logger.debug('Removing any stale branches');
   logger.trace({ config }, `pruneStaleBranches`);
   logger.debug(`config.repoIsOnboarded=${config.repoIsOnboarded}`);
-  if (!branchList) {
+  if (!branchList || branchList.length === 0) {
```

Comment:
🤔 why the branch list can now be empty? why we need to handle this explicit?

Reply:
had to be added as without it any new config migration branch is being auto closed in the same go, as the branch list is empty therefore all the filters in prune are not executed and we end up auto closing